native/c/test/uss.test.cpp

```diff
@@ -0,0 +1,581 @@
+                                        string getPermissionsCommand = "ls -l " + uss_path;
+                                        string chmodFileCommand = zowex_command + " uss chmod 777 " + uss_path;
+
+                                        rc = execute_command_with_output(getPermissionsCommand, response);
+                                        ExpectWithContext(rc, response).ToBe(0);
+                                        Expect(response).ToContain(("-rw-r--r--"));
+
+                                        rc = execute_command_with_output(chmodFileCommand, response);
+                                        ExpectWithContext(rc, response).ToBe(0);
+                                        rc = execute_command_with_output(getPermissionsCommand, response);
+                                        ExpectWithContext(rc, response).ToBe(0);
+                                        Expect(response).ToContain(("-rwxrwxrwx"));
+                                      });
+                                 });
+
+                        describe("on a directory",
+                                 [&]() -> void
+                                 {
+                                   beforeEach([&]() -> void
+                                              {
+                                                uss_path = get_random_uss(ussTestDir) + "_dir";
+                                                create_test_dir_cmd(uss_path); });
+
+                                   it("should properly chmod a directory",
+                                      [&]() -> void
+                                      {
+                                        string getPermissionsCommand = "ls -ld " + uss_path;
+                                        string chmodFileCommand = zowex_command + " uss chmod 777 " + uss_path + " -r";
+
+                                        rc = execute_command_with_output(getPermissionsCommand, response);
+                                        ExpectWithContext(rc, response).ToBe(0);
+                                        Expect(response).ToContain(("drwxr-xr-x"));
+
+                                        rc = execute_command_with_output(chmodFileCommand, response);
+                                        ExpectWithContext(rc, response).ToBe(0);
+                                        rc = execute_command_with_output(getPermissionsCommand, response);
+                                        ExpectWithContext(rc, response).ToBe(0);
+                                        Expect(response).ToContain(("drwxrwxrwx"));
+                                      });
+                                 });
+
+                        it("should properly handle calling chmod on a file that does not exist",
+                           [&]() -> void
+                           {
+                             string uss_file = ussTestDir + "/test_does_not_exist";
+                             string response;
```

Comment:
This `response` variable can be removed as its a redundant declaration (shadows outer scope on line 30)